src/main/kotlin/com/devfloor/untitled/articlehashtag/application/ArticleHashtagService.kt

```diff
@@ -11,6 +12,18 @@ class ArticleHashtagService(
     private val repository: ArticleHashtagRepository
 ) {
     @Transactional(readOnly = true)
-    fun showAllByArticle(article: Article): List<ArticleHashtag> =
-        repository.findAllByArticle(article)
+    fun showAllByArticle(article: Article): List<ArticleHashtag> {
+        return repository.findAllByArticle(article)
+    }
+
+    @Transactional
+    fun create(article: Article, hashtagList: List<Hashtag>): List<ArticleHashtag> {
+        val articleHashtagList = mutableListOf<ArticleHashtag>()
+        for(index in hashtagList.indices){
+            articleHashtagList.add(repository.save(ArticleHashtag(article, hashtagList[index])))
+        }
+        return articleHashtagList
+    }
+
```

Comment:
이와 같이 변경하면 코드가 더 간단해 질 수도 있을거같아요 !!
아래와 같이 작성하면 코드가 더 간단해질 수 있을 것 같아요 !
```suggestion
        hashtagList.forEach {
            articleHashtagList.add(repository.save(ArticleHashtag(article, it)))
        }
```

Reply:
아 그런데, 해당 create 메서드를 이용하는 facade에서는 create의 반환값인 `List<ArticleHashtag>` 를 받아서 할당만 해놓지, 정작 그 변수를 이용하지는 않는 것으로 보입니다 !
따라서 아무것도 반환하지 않는 형태로 바꾸고, 21번째 줄과 25번째 줄을 생략해도 되지 않을까요!?

Reply:
제가 줄을 잘못 insert했네요ㅠ 여기서 commit suggestion을 누르지 않고, 직접 변경하셔야 될 것 같아요 ㅠ

Reply:
여기서도 for문을 forEach문으로 변경하는 것이 항상 좋은 방법인가에 대해 알아보면 좋을 것 같아요!

[for-loop 를 Stream.forEach() 로 바꾸지 말아야 할 3가지 이유](https://homoefficio.github.io/2016/06/26/for-loop-%EB%A5%BC-Stream-forEach-%EB%A1%9C-%EB%B0%94%EA%BE%B8%EC%A7%80-%EB%A7%90%EC%95%84%EC%95%BC-%ED%95%A0-3%EA%B0%80%EC%A7%80-%EC%9D%B4%EC%9C%A0/)

Reply:
제가 생각했을 때 for과 forEach의 가장 큰 차이는
iter 돌다가 break나 continue를 통해 루프를 빠져나오는 것이 필요하냐 혹은 해당 list를 한바퀴 전부 도는 것이 목적이냐에 따라 다른 것 같아요.
그런데 현재 코드에서는 list 한바퀴를 전체 돌면서 index를 통해 각 element에 접근하고 있는데, 중간에 빠져나오는 것이 필요한게 아니라면 forEach가 좀더 적절한 코드라고 생각합니다.

Reply:
아 그리고 성능에 대해서도, 일반적인 반복문이 아닌 list, map과 같은 컬렉션을 도는 부분이라면 forEach가 더 빠르다고도 합니다 !

Reply:
요건 forEach 구현부 입니다. 보니까 얘도 내부적으로는 for로 구현되어있네요
![image](https://user-images.githubusercontent.com/49271821/116571787-5b547a80-a946-11eb-94f5-5314d1a6515d.png)

Reply:
오 Java에서 stream을 통한 forEach랑 코틀린이랑 조금 차이가 있네요!! 좋은 정보 감사함다 ㅎㅅㅎ